pkgs/build-support/binary-cache/make-binary-cache.py

```diff
@@ -1,43 +1,78 @@
-
+from functools import partial
 import json
+from multiprocessing import Pool
 import os
 import subprocess
 
-with open(os.environ["NIX_ATTRS_JSON_FILE"], "r") as f:
-  closures = json.load(f)["closure"]
 
-os.chdir(os.environ["out"])
+def dropPrefix(path, nixPrefix):
+    return path[len(nixPrefix + "/") :]
+
+
+def processItem(item, nixPrefix, outDir):
+    os.chdir(outDir)
+
+    narInfoHash = dropPrefix(item["path"], nixPrefix).split("-")[0]
+
+    xzFile = "nar/" + narInfoHash + ".nar.xz"
+    with open(xzFile, "w") as f:
```

Comment:
I think this wants binary mode

```suggestion
    with open(xzFile, "wb") as f:
```